formbricks/formbricks
# feat: enable weekly summary & support for callbacks on login

## What does this PR do?
- Enables Weekly Summaries to be on by default for new users & new products
- Adds support for a `callbackUrl` searchParam on login endpoint to allow for redirection to specific pages just after login (will help in future as well)
    - This led to the removal of the redirect to `/` from the auth/`layout.tsx` because next does not allow reading params/search params in the server side layouts. It only allows in server side pages
    - Hence, all the manual redirects had to be put in place for the cases where they already had an active session
- Adds `notificationSettings` in the user type that is returned from all the user services.
- Adds the following snippet of  code after every product creation to add it to the weekly summary:
```ts
  const updatedNotificationSettings = {
    ...session.user.notificationSettings,
    alert: {
      ...session.user.notificationSettings?.alert,
    },
    weeklySummary: {
      ...session.user.notificationSettings?.weeklySummary,
      [product.id]: true,
    },
  };

  await updateUser(session.user.id, {
    notificationSettings: updatedNotificationSettings,
  });
```
- Replaces the email footer of weekly summary from:
```html
This is a Beta feature. If you experience any issues, please let us know by replying to this email 🙏
```
to 
```html
To halt Weekly Updates, <a href="${WEBAPP_URL}/auth/login?callbackUrl=${WEBAPP_URL}/environments/${notificationData.environmentId}/settings/notifications">please turn them off</a> in your settings 🙏
```

- This is how the email footer looks like now to wrap it inside 1 line:
![image](https://github.com/formbricks/formbricks/assets/55556994/4fe13822-d9be-4f50-8697-9deadfe30f2f)


## How should this be tested?
Create new user/product and then check the notification settings, they should be enabled for weekly summary.
Hit the weekly summary cron endpoint manually and verify the email footer, the redirect on signed up as well as not signed up users!

## Checklist



### Required


### Appreciated
